test/create-sys-script.py

```diff
@@ -0,0 +1,202 @@
+#!/usr/bin/python3
+
+OUTFILE_HEADER = u"""#!/usr/bin/python3
+#
+# create-sys-script.py
+#
+# (C) 2017 Canonical Ltd.
+# Author: Dan Streetman <dan.streetman@canonical.com>
+#
+# systemd is free software; you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation; either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# systemd is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with systemd; If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+
+# Use this only to (re-)create the test/sys-script.py script,
+# after adding or modifying anything in the test/sys/ directory
+
+
+import os, sys, stat, tempfile, filecmp
+
+
+OUTFILE = u"sys-script.py"
+OUTFILE_MODE = stat.S_IRWXU | stat.S_IRWXG | stat.S_IROTH | stat.S_IXOTH
+
+OUTFILE_FUNCS = r"""
+import os, sys
+
+def d(path, mode):
+    os.mkdir(path)
+    os.chmod(path, int(mode, 8))
```

Comment:
> os.mkdir has a mode argument, so you can replace d by a single mkdir call.

yes, just trying to go by the docs which say:
"On some systems, mode is ignored"
however as this will only ever be run on unix (i assume) I'll fold the mode into the mkdir() call

> Python supports octal values (0o644) which make the intent much more clear

yeah, just trying to be "pythony" instead of "unixy", but again as this is always run on unix using the octal directly is clearer.